src/vstest.console/Processors/RunSettingsArgumentProcessor.cs

```diff
@@ -135,16 +136,32 @@ public void Initialize(string argument)
                 this.runSettingsManager.AddDefaultRunSettings();
 
                 this.commandLineOptions.SettingsFile = argument;
+
+                // runsettings with data collector and testsettings is not supported.
```

Comment:
We should move this to a common place for both CLI and IDE scenarios.

Reply:
Moved in TestRequestManager.cs